yaml/yaml_test.go

```diff
@@ -113,3 +113,38 @@ func TestArray(t *testing.T) {
 	assert.Equal(t, verify[0]["c"], 3)
 	assert.Equal(t, verify[1]["c"], 4)
 }
+
+type hasVars struct {
+	Vars []variable
+}
+
+type variable struct {
+	Default interface{}
+}
+
+func TestStructEmptyArray(t *testing.T) {
+	input := []byte(`
+vars:
+  - default: []
+  - default: null
+  - default: 3
+  - default:
+    - a
+    - b
+    - c
+`)
+
+	c, err := NewConfig(input)
+	if err != nil {
+		t.Fatalf("failed to parse input: %v", err)
+	}
+
+	var verify hasVars
+	err = c.Unpack(&verify)
+	assert.Nil(t, err)
+
+	assert.Nil(t, verify.Vars[0].Default)
+	assert.Nil(t, verify.Vars[1].Default)
+	assert.Equal(t, uint64(3), verify.Vars[2].Default)
+	assert.Len(t, verify.Vars[3].Default.([]interface{}), 3)
```

Comment:
Can we split this up into 4 test cases (e.g. table driven test), or is the error only triggered by this setup?

Was the error triggered for each of these cases?

Reply:
Let's add a sample test with an empty array as has been shown in the issue.

Reply:
Good catch. It seems that an empty array is also an issue. I simplified the test case.